### PR TITLE
chore: custom lint rules

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

The warning lint rules for `react/forbid-elements` are not applied because of a duplicate key in the object. This means that the warnings never appear in files

## Changes

- Add support for custom linting rules
- Duplicate the `react/forbid-elements` rule and rename it `posthog/warn-elements`

Based on the great work from @pauldambra in https://github.com/PostHog/posthog-js/pull/843

## How did you test this code?

|Before|After|
|----|----|
| Screenshot 2023-10-26 at 11 24 07 | Screenshot 2023-10-26 at 11 21 55 |

### CODE REVIEW

File: eslint-rules/warn-elements.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const { rules } = require('eslint-plugin-react')
+
+module.exports = rules['forbid-elements']
```
Comment (pauldambra) [inline eslint-rules/warn-elements.js:3]: noice! 💯

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.1'
+lockfileVersion: '6.0'
```
Comment (pauldambra) [inline pnpm-lock.yaml:1]: pnpm is opaque to me... do we care about this downgrade?

**Review by pauldambra — APPROVED** [review]

File: pnpm-lock.yaml (hunk 2 above)
Comment (neilkakkar) [reply pnpm-lock.yaml:1]: don't worry, a random PR from me will upgrade it soon anyway ☠️ , and the cycle will continue

Comment (daibhin) [reply pnpm-lock.yaml:1]: I can't find the exact link but I came across this in another thread recently and the tldr was that it just flapped based on a local pnpm version. Not majorly important

**Pull request merged**